tests/integration_tests/build/test_binary_size.py

```diff
@@ -13,9 +13,9 @@
 
 SIZES_DICT = {
     "x86_64": {
-        "FC_BINARY_SIZE_TARGET": 2367352,
+        "FC_BINARY_SIZE_TARGET": 2520632,
         "JAILER_BINARY_SIZE_TARGET": 797632,
-        "FC_BINARY_SIZE_LIMIT": 2485719,
+        "FC_BINARY_SIZE_LIMIT": 2646663,
```

Comment:
This increase is significant (2646663 − 2485719 = 160944 bytes (~157.7 KB)). This equates to roughly 6% increase in the binary size. I'm assuming that is all due to `thiserror` being added as a dependency.

Reply:
`thiserror` is already included as a transitive dependency via `userfaultfd` and `vm-allocator` so this wouldn't contribute additionally to the binary size. The increase is a result of the static strings uses for the `std::fmt::Display` implementations for the error types and some of the additional code generated with `std::error::Error`.

Reply:
The use of a library in a dependency wouldn't automatically result in the library being linked in. I would have expected the net delta in the binary size as a result of the code change in the immediate repository to be better not worse. The change here should be entirely related to the use of `thiserror` locally.

Regardless, as discussed offline, the change to the size target/limit here is not exclusively owed to the use of `thiserror` due to the 5% tolerance margin. It is just that the change here finally breached the threshold.

Reply:
I ran the `test_binary_size.py` locally and came up with a change of `(2520632 - 2455280) = 65352` which is less serious.